src/LinAlg/hiopMatrixDenseRowMajor.cpp

```diff
@@ -787,6 +787,48 @@ double hiopMatrixDenseRowMajor::max_abs_value()
   return maxv;
 }
 
+void hiopMatrixDenseRowMajor::row_max_abs_value(hiopVector &ret_vec)
+{
+  char norm='M';
+  int one=1;
+  double maxv;
+  
+  hiopVectorPar& vec=dynamic_cast<hiopVectorPar&>(ret_vec);
+  assert(m_local_==vec.get_local_size());
+  
+  for(int irow=0; irow<m_local_; irow++)
+  {
+    maxv = DLANGE(&norm, &one, &n_local_, M_[0]+(irow*n_local_), &one, nullptr);
+#ifdef HIOP_USE_MPI
+    double maxvg;
+    int ierr=MPI_Allreduce(&maxv,&maxvg,1,MPI_DOUBLE,MPI_MAX,comm_); assert(ierr==MPI_SUCCESS);
```

Comment:
Wouldn't be better to do a single `MPI_Allreduce` (of `vec`) after the for loop?